include/mata/nfa/delta.hh

```diff
@@ -121,64 +121,105 @@
 
     /**
-     * Iterator over moves. It iterates over pairs (symbol, target state) for a current source state whose state post
-     *  we iterate.
+     * @brief Iterator over moves represented as @c Move instances.
+     *
+     * It iterates over pairs (symbol, target) for the given @c StatePost.
      */
-    struct moves_const_iterator {
-    private:
-        const std::vector<SymbolPost>& symbol_posts_{};
-        std::vector<SymbolPost>::const_iterator symbol_post_it_{};
-        StateSet::const_iterator target_states_it_{};
-        bool is_end_{ false };
-        Move move_{};
-
-    public:
-        using iterator_category = std::forward_iterator_tag;
-        using value_type = Move;
-        using difference_type = unsigned;
-        using pointer = Move*;
-        using reference = Move&;
-
-        explicit moves_const_iterator(const std::vector<SymbolPost>& symbol_posts, bool is_end = false);
-
-        moves_const_iterator(const std::vector<SymbolPost>& symbol_posts,
-                             std::vector<SymbolPost>::const_iterator symbol_posts_it,
-                             StateSet::const_iterator target_states_it, bool is_end = false);
-
-        moves_const_iterator(const moves_const_iterator& other) = default;
-
-        const Move& operator*() const { return move_; }
-
-        // Prefix increment
-        moves_const_iterator& operator++();
-        // Postfix increment
-        const moves_const_iterator operator++(int);
-
-        moves_const_iterator& operator=(const moves_const_iterator& x);
-
-        bool operator==(const moves_const_iterator& other) const;
-        bool operator!=(const moves_const_iterator& other) const { return !(*this == other); };
-    };
-
-    moves_const_iterator moves_cbegin() const { return moves_const_iterator(ToVector()); }
-    moves_const_iterator moves_cend() const { return moves_const_iterator(ToVector(), true); }
-    moves_const_iterator moves_begin() const { return moves_cbegin(); }
-    moves_const_iterator moves_end() const { return moves_cend(); }
-
     class Moves {
     public:
-        moves_const_iterator begin_;
-        moves_const_iterator end_;
-        moves_const_iterator begin() const { return begin_; }
-        moves_const_iterator end() const { return end_; }
-    };
+        /**
+         * @brief Whether to look up first symbol to iterate over from the beginning or from the end of @c StatePost
+         *  iterator.
+
+         * Iterates over symbol posts with symbols in interval [first_symbol_, last_symbol_].
+         */
+        enum class FirstSymbolLookupDiretion {
+            Forward, ///< Lookup from the beginning to the end.
+            Backward ///< Lookup from the end to the beginning.
+        };
+
+        /**
+         * Iterator over moves.
+         */
+        class const_iterator {
+        private:
+            const StatePost* state_post_{ nullptr };
+            StatePost::const_iterator state_post_it_{};
```

Comment:
You either name the iterators after whose iterator it is (as in this case) or over what data the iterator iterates (in this case, it would be `symbol_post_it` iterating over symbol posts). I am fine with either, and I think both approaches are used equally. If you both think it would be better, I am all for changing the name as you suggest.

Reply:
It was decided that we will name iterators as `<the_name_of_the_object_we_get_by_dereferencing_the_iterator>_it`. We can slowly begin renaming all existing iterator variables to adhere to this convention.

For example, for const iterator over `StatePost`, we get the following variable name:
```cpp
StatePost::const_iterator symbol_post_it = nfa.delta.state_post(4).begin();
```

Reply:
For further discussion, please refer to https://github.com/VeriFIT/mata/issues/313.